### ISSUE TITLE
Rate limiting breaks annotation UI

### ISSUE THREAD

**Issue by pratclot** [cvat-ai/cvat#8805]
### Actions before raising this issue

- [X] I searched the existing issues and did not find anything similar.
- [X] I read/searched [the docs](https://docs.cvat.ai/docs/)

### Steps to Reproduce

This is not exactly a reproducer, just what I did to "break" the server :)
1. Upload 19 4k .png files to a project.
2. Annotate ~7 of them.
3. Be unable to move to remaining ones.

### Expected Behavior

Everything works by default :)

### Context

Hey team, I am stuck with a `uvicorn` error when trying to annotate images, here is a line from `cvat-backend` logs:
```bash
2024-12-09 18:43:25,606 DEBG 'uvicorn-0' stdout output:
INFO:     10.15.0.40:0 - "GET /api/jobs/1/data?org=&quality=compressed&type=chunk&index=0 HTTP/1.0" 429 Too Many Requests
```
I am not sure why there are too many requests to be honest, I am just a single user trying to annotate pictures through the UI.

### Environment

```Markdown
Using provided Helm chart from 12eec5481bb3d4e6219bb8214df524f27b3037e3.
```

**Comment by bsekachev** [comment]
Since 2.23.0 main server processes does not prepare data chunks anymore. The process is done in dedicated workers.

Make sure you have at least one active pod `cvat_worker_chunks`.

It is not really intuitive, hovewer in this specific case the code means, that this worker was not able to prepare chunk in one minute. It may refer to different things (you do not have such workers or all these workers are busy).

That approach was chosen to keep API compatibility.

**Comment by bsekachev** [comment]
Related https://github.com/cvat-ai/cvat/issues/8781

**Comment by bsekachev** [comment]
May be there is something useful in pod logs?

**Comment by pratclot** [comment]
Not really, but then again I would not know which pod is important for this (the `chunks` one only has this traceback repeated over and over again, the `server` only the 429 and some 200 responses, others also do not seem to have important clues).

I switched to `v2.22.0` though, it seems to work, lol.

**Comment by sriramsowmithri9807** [comment]
Hi team,

I’m encountering a similar issue when trying to annotate images in CVAT, and I wanted to share my experience and findings.

Actions Before Raising This Issue
I searched through existing issues and did not find anything similar.
I reviewed the documentation for any relevant information.

### Steps to Reproduce
While this isn't a strict reproducer, here’s what I did that led to the issue:

Uploaded 19 4K .png files to a project.
Annotated approximately 7 of them.
Found myself unable to move to the remaining images.
Expected Behavior
I expected everything to work seamlessly by default.
### 
Context
I am facing a uvicorn error when trying to annotate images. Here’s a relevant line from the cvat-backend logs:


Verify

Open In Editor
Run
Copy code
```
2024-12-09 18:43:25,606 DEBG 'uvicorn-0' stdout output:
INFO:     10.15.0.40:0 - "GET /api/jobs/1/data?org=&quality=compressed&type=chunk&index=0 HTTP/1.0" 429 Too Many Requests
```
It’s puzzling because I’m just a single user trying to annotate images through the UI, and I’m not sure why there are too many requests.

Environment
I’m using the provided Helm chart from commit 12eec5481bb3d4e6219bb8214df524f27b3037e3.

Additional Information
I noticed that since` version 2.23.0`, the main server processes no longer prepare data chunks; this is now handled by dedicated workers. I made sure that at least one cvat_worker_chunks pod is active, but it seems that the worker is unable to prepare chunks in a timely manner, possibly due to being busy or not existing.

I also encountered a` PermissionError in the cvat-backend-worker-chunks logs,` indicating that it cannot create directories due to permission issues:


Verify

Open In Editor
Run
```
Copy code
PermissionError: [Errno 13] Permission denied: '/home/django/data/cache/events'
```
After reinstalling the chart and removing all volume claims and the namespace in Kubernetes, I still see these errors. I even tried setting the access modes to ReadWriteMany for the cvat-kvrocks-data and cvat-backend-data PVCs, but that did not resolve the issue.

I switched back to version 2.22.0, and it seems to work without these problems.

If anyone has insights or suggestions on how to resolve this, I would greatly appreciate it. Thank you for your help!

Best,
Sowmithri Sriram.
sriramsowmithri9807 --> github

**Comment by CarlQLange** [comment]
One way to fix this is to set the pod `securityContext.runAsUser: 0`, `kubectl exec` into the pod, `chown -R django:django data/cache && chmod -R 777 data/cache`, and then undo the securityContext again. IIRC the traditional way to do this is to have a fixPermissions sidecare.
